src/contentscript/overlay/root/components/SystemPopup/components/Session/index.tsx

```diff
@@ -14,8 +14,11 @@ export const Session: FC<SessionProps> = (p: SessionProps) => {
 
           <div className={styles.info}>
             <ul className={styles.icons}>
-              {p.providerIcon ? <img src={p.providerIcon} className={styles.icon} /> : null}
-              {p.walletIcon ? <img src={p.walletIcon} className={styles.icon} /> : null}
+              {p.providerIcon ? (
+                <img src={p.providerIcon} className={styles.icon} />
+              ) : p.walletIcon ? (
+                <img src={p.walletIcon} className={styles.icon} />
+              ) : null}
```

Comment:
In the case of wallet connect, the icons will be different and it is important for us to know that it is, for example, a Trust Wallet connected via WalletConnect. 

Here we need to change the condition:
If the icons are the same, then show one icon. 
If the icons are different, then show two icons. 

I think a simple comparison (providerIcon === walletIcon) should work.